sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -1258,3 +1242,71 @@ def test_gaussian_mixture_setting_best_params():
         "lower_bound_",
     ]:
         assert hasattr(gmm, attr)
+
+
+@pytest.mark.parametrize(
+    "init_params", ["random", "random_from_data", "k-means++", "kmeans"]
+)
+def test_init_param(init_params):
+    # Check that the init_param options produces valid output.
+    # Compare all to default (kmeans).
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+
+    n_components = rand_data.n_components
+    X = rand_data.X["full"]
+    gmm = GaussianMixture(
+        n_components=n_components, random_state=rng, init_params="kmeans"
+    )
+    gmm.fit(X)
+    default_means = np.sort(gmm.means_.flatten())
+
+    gmm = GaussianMixture(
+        n_components=n_components,
+        random_state=rng,
+        init_params=init_params,
+        tol=1e-06,
+        max_iter=1000,
+    )
+    gmm.fit(X)
+    assert_almost_equal(default_means, np.sort(gmm.means_.flatten()), decimal=10)
```

Comment:
It's better to use `assert_allclose` which checks relative difference

Reply:
I used ```assert_allclose``` with ```rtol=1e-8```. Does that sound good or should I constrain it a bit more?

Reply:
Actually you can leave rtol to its default value, which is 1e-7. This is what we consider acceptable throughout the project.